### PR TITLE
Setting up check-style and spot-bugs

### PR DESCRIPTION
**Pull request by rasika**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: gradle/javaProject.gradle [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    testCompile 'org.testng:testng:6.13.1'
+
+    checkstyle project(":build-config:checkstyle")
+    checkstyle "com.puppycrawl.tools:checkstyle:${puppycrawlCheckstyleVersion}"
+}
+
+test {
+    systemProperty "java.util.logging.config.file", "src/test/resources/logging.properties"
+    systemProperty "java.util.logging.manager", "org.ballerinalang.logging.BLogManager"
+    systemProperty "ballerina.home", "/home/miyurud/software/ballerina-1.2.4/distributions/jballerina-1.2.4"
```
Comment (hemikak) [inline gradle/javaProject.gradle:26]: Will this break the tests ?

File: .gitignore [hunk 2]
```diff
@@ -11,13 +11,38 @@
...
+
+# MacOS
+*.DS_Store
+
+#Ballerina
+*.balx
+samples/*/kubernetes
+samples/*/docker
+velocity.log*
+*Ballerina.lock
```
Comment (hemikak) [inline .gitignore:48]: Shall we remove unwanted entries ?

Comment (rasika) [reply .gitignore]: +1, added a fix

File: gradle/javaProject.gradle (hunk 1 above)
Comment (rasika) [reply gradle/javaProject.gradle:26]: @miyurud will be working on this, just copied test-task from his build gradle.

**Comment by rasika** [comment]
@hemikak shall we address the changes for the "ballerina.home" and the rest in a separate PR. This PR only enables check-style plugin and there-are many more check-style failures to fix as well.

File: gradle/javaProject.gradle (hunk 1 above)
Comment (miyurud) [reply gradle/javaProject.gradle:26]: Yes, I will fix this issue

**Comment by hemikak** [comment]
@rasika , ok sure.

**Pull request merged**